app/DynamicGraphs/GraphGenerator.hs

```diff
@@ -22,49 +22,61 @@ import qualified Data.Map.Strict as Map
 import Database.Requirement (Req(..))
 import Data.Sequence as Seq
 import Data.Hash.MD5 (Str(Str), md5s)
-import Data.Text.Lazy (Text, pack, isInfixOf)
+import Data.Text.Lazy (Text, pack, unpack, isInfixOf, isPrefixOf)
 import Data.Containers.ListUtils (nubOrd)
 import Control.Monad.State (State)
 import qualified Control.Monad.State as State
 import Control.Monad (mapM, liftM)
-
+import DynamicGraphs.GraphOptions (GraphOptions(..))
 
 -- | Generates a DotGraph dependency graph including all the given courses and their recursive dependecies
 coursesToPrereqGraph :: [String] -- ^ courses to generate
                         -> IO (DotGraph Text)
-coursesToPrereqGraph = coursesToPrereqGraphExcluding []
+coursesToPrereqGraph rootCourses = coursesToPrereqGraphExcluding (getDefaultGraphOptions (map pack rootCourses))
 
 -- | Takes a list of taken courses, along with a list of courses we wish to generate
 -- a dependency graph for. The generated graph will neither include any of the taken courses,
 -- nor the dependencies of taken courses (unless they are depended on by other courses)
-coursesToPrereqGraphExcluding :: [String] -- ^ taken courses
-                              -> [String] -- ^ course to generate
+coursesToPrereqGraphExcluding :: GraphOptions
                               -> IO (DotGraph Text)
-coursesToPrereqGraphExcluding taken courses = do
-    reqs <- lookupCourses taken $ map pack courses
+coursesToPrereqGraphExcluding options = do
+    reqs <- lookupCourses (map unpack (taken options)) (courses options)
     let reqs' = Map.toList reqs
-    return $ fst $ State.runState (reqsToGraph reqs') initialState
+    return $ fst $ State.runState (reqsToGraph options reqs') initialState
     where
         initialState = GeneratorState 0 Map.empty
 
 sampleGraph :: DotGraph Text
-sampleGraph = fst $ State.runState (reqsToGraph [
-    ("MAT237H1", J "MAT137H1" ""),
+sampleGraph = fst $ State.runState (reqsToGraph
+    (getDefaultGraphOptions ["MAT237H1", "MAT133H1", "CSC148H1", "CSC265H1"])
+    [("MAT237H1", J "MAT137H1" ""),
     ("MAT133H1", NONE),
     ("CSC148H1", AND [J "CSC108H1" "", J "CSC104H1" ""]),
     ("CSC265H1", AND [J "CSC148H1" "", J "CSC236H1" ""])
     ])
     (GeneratorState 0 Map.empty)
 
+getDefaultGraphOptions :: [Text] -> GraphOptions
```

Comment:
Two main comments here:

1. Typically we'd put a "default" value with the data type definition itself (i.e.,g in the `GraphOptions` file.
2. Don't make `rootCourses` part of the options---notice that it's fundamentally a separate argument to the `reqsToStmts` function. Think of `GraphOptions` as only configuring the look and structure of the resulting graph.

Reply:
Thanks David! I removed courses from GraphOptions and made a separate branch for a [separate PR](https://github.com/Courseography/courseography/pull/1163) to introduce the GraphOptions type.